src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
+#include "software/ai/hl/stp/tactic/kick/kick_fsm.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/intent.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/ai/passing/pass.h"
+#include "software/geom/algorithms/closest_point.h"
+
+struct ReceiverFSM
+{
+    class OneTouchShotState;
+    class ReceiveAndDribbleState;
+    class UndecidedState;
+
+    struct ControlParams
+    {
+        // The pass to receive
+        std::optional<Pass> pass = std::nullopt;
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    // The minimum proportion of open net we're shooting on vs the entire size of the net
+    // that we require before attempting a shot
+    static constexpr double MIN_SHOT_NET_PERCENT_OPEN    = 0.3;
+    static constexpr double MAX_SPEED_FOR_ONE_TOUCH_SHOT = 6.5;
+    static constexpr double MIN_PASS_START_SPEED         = 1.0;
+
+    // The maximum deflection angle that we will attempt a one-touch kick towards the
+    // enemy goal with
+    static constexpr Angle MAX_DEFLECTION_FOR_ONE_TOUCH_SHOT = Angle::fromDegrees(90);
+
+    /**
+     * Given a shot and the ball, figures out the angle the robot should be facing
+     * to perform a one-touch shot.
+     *
+     * @param shot The shot to take
+     * @param ball The ball on the field
+     */
+    static Angle getOneTimeShotDirection(const Ray& shot, const Ball& ball)
```

Comment:
nit: could this be called getOneTouchShotDirection?

Reply:
updated